### DISCUSSION TITLE
How to delete the current line.

### DISCUSSION THREAD

**Discussion by liukaizheng** [helix-editor/helix#705] [Q&A] [answered]
The command is 'dd' in vim, which I often used. But I can't find similar command in helix.

**Comment by sudormrfbin** [comment, marked as answer]
You can use `xd`, which will select the current line (`x`) and delete it (`d`). Tip: pressing `x` repeatedly will select more lines downwards.

**Reply by ellbur** [reply]
When used on a blank line, `x` seems to select the next line. Is there a way to delete a line that works the same for blank and non-blank lines?

**Reply by ellbur** [reply]
It seems `Xd` might do the trick.

**Reply by fadyosman** [reply]
@ellbur for blank lines d by itself will do the trick, no need to do Xd.

**Reply by ellbur** [reply]
@fadyosman good point, though then I have to think "Is this line bank?" That's too much thinking for me.

**Comment by Dialga** [comment]
how can one map `dd` to delete the current line?

**Reply by sync2cloud** [reply]
add this to your `~/.config/helix/config.toml
`
```
[keys.normal]
d = { d = ["extend_line_below", "delete_selection"] }
```

**Reply by weakish** [reply]
I would use `extend_to_line_bounds` instead of `extend_line_below`. Although this still just handle the simple case of `delete current line`. Things like `5dd` to delete 5 lines in vim are not supported. Also, this makes pressing `d` to enter a minor mode, helix will wait for the second key press. The original function (delete selections when pressing `d`) is lost.

**Reply by Zdertis420** [reply]
And what if I have ```d``` mapped to one action and I want to map ```dd``` for another?
